Realm/RLMSyncUtil_Private.h

```diff
@@ -22,6 +22,17 @@
 #import <Realm/RLMRealmConfiguration.h>
 #import <Realm/RLMSyncCredentials.h>
 
+typedef NS_ENUM(NSUInteger, RLMSyncSystemErrorKind) {
```

Comment:
`asyncOpen` can yield an error with an `RLMSyncSystemErrorKind` if the error is produced by the sync subsystem.